src/clu_main.c

```diff
@@ -0,0 +1,221 @@
+               "To resolve please recompile wolfSSL with the correct integrity"
+               "hash. If the issue continues, contact fips @ wolfssl.com");
+    }
+#endif
+
+    /* If the first string does not have a '-' in front of it then try to
+     * get the mode to use i.e. x509, req, version ... this is for
+     * compatibility with the behavior of the OpenSSL command line utility
+     */
+    if (argc > 1 && argv[1] != NULL && argv[1][0] != '-') {
+        flag = getMode(argv[1]);
+    }
+    else {
+        /* retain old version of modes where '-' is used. i.e -x509, -req */
+        flag = getopt_long_only(argc, argv,"", mode_options, &longIndex);
+    }
+
+    switch (flag) {
+        case 0:
+            WOLFCLU_LOG(WOLFCLU_L0, "No mode provided.");
+            ret = 0;
+            break;
+
+        case WOLFCLU_CRYPT:
+            /* generic 'enc' used, default to encrypt unless -d was used */
+            ret = wolfCLU_checkForArg("-d", 2, argc, argv);
+            if (ret > 0) {
+                ret = wolfCLU_setup(argc, argv, 'd');
+            }
+            else {
+                ret = wolfCLU_setup(argc, argv, 'e');
+            }
+            break;
+
+        case WOLFCLU_ENCRYPT:
+            ret = wolfCLU_setup(argc, argv, 'e');
+            break;
+
+        case WOLFCLU_DECRYPT:
+            ret = wolfCLU_setup(argc, argv, 'd');
+            break;
+
+        case WOLFCLU_BENCHMARK:
+            ret = wolfCLU_benchSetup(argc, argv);
+            break;
+
+        case WOLFCLU_HASH:
+            ret = wolfCLU_hashSetup(argc, argv);
+            break;
+
+        case WOLFCLU_MD5:
+            ret = wolfCLU_md5Setup(argc, argv);
+            break;
+
+        case WOLFCLU_X509:
+            ret = wolfCLU_certSetup(argc, argv);
+            break;
+
+        case WOLFCLU_REQUEST:
+            ret = wolfCLU_requestSetup(argc, argv);
+            break;
+
+        case WOLFCLU_GEN_KEY:
+            ret = wolfCLU_genKeySetup(argc, argv);
+            break;
+
+        case WOLFCLU_ECPARAM:
+            ret = wolfCLU_ecparam(argc, argv);
+            break;
+
+        case WOLFCLU_PKEY:
+            ret = wolfCLU_pKeySetup(argc, argv);
+            break;
+
+        case WOLFCLU_DGST:
+            ret = wolfCLU_dgst_setup(argc, argv);
+            break;
+
+        case WOLFCLU_RSA:
+        case WOLFCLU_ECC:
+        case WOLFCLU_ED25519:
+            ret = wolfCLU_sign_verify_setup(argc, argv);
+            break;
+
+        case WOLFCLU_HELP:
+            /* only print for -help if no mode has been declared */
+            WOLFCLU_LOG(WOLFCLU_L0, "Main help menu:");
+            wolfCLU_help();
+            return 0;
+
+        case WOLFCLU_VERBOSE:
+            wolfCLU_verboseHelp();
+            return 0;
+
+        case 'v':
+            if (wolfCLU_version() != 0) {
+                WOLFCLU_LOG(WOLFCLU_L0, "Error getting version");
+            }
+            return FATAL_ERROR;
```

Comment:
Should this be `return 0`? And `return FATAL_ERORR` only if `wolfCLU_version()` failed (i.e. in the conditional)? I still think `wolfCLU_version` should have a void return, though, which would make these comments irrelevant. I don't see how that function could meaningfully fail, and as it stands, it always returns 0.

Reply:
reworked a lot of the error codes throughout wolfCLU and fixed the return value here